lib/spack/spack/solver/concretize.lp

```diff
@@ -601,6 +644,10 @@ node_target_mismatch(Parent, Dependency)
   :- depends_on(Parent, Dependency),
      not node_target_match(Parent, Dependency).
 
+% disallow reusing concrete specs that don't have a compatible target
+:- node(Package), node_target(Package, Target), not target(Target),
+   error("No available package's target was compatible with this machine").
```

Comment:
I think this should be "No satisfying package's target is compatible with this machine". The space of all packages/all available packages is probably too broad

Reply:
addressed